packages/core/contracts/financial-templates/perpetual-multiparty/PerpetualCreator.sol

```diff
@@ -120,4 +134,15 @@ contract PerpetualCreator is ContractCreator, Testable, Lockable {
         constructorParams.liquidationLiveness = params.liquidationLiveness;
         constructorParams.excessTokenBeneficiary = params.excessTokenBeneficiary;
     }
+
+    // IERC20Standard.decimals() will revert if the collateral contract has not implemented the decimals() method,
+    // which is possible since the method is only an OPTIONAL method in the ERC20 standard:
+    // https://eips.ethereum.org/EIPS/eip-20#methods.
+    function _getSyntheticDecimals(address _collateralAddress) public view returns (uint8 decimals) {
+        try IERC20Standard(_collateralAddress).decimals() returns (uint8 _decimals) {
+            return _decimals;
+        } catch {
+            return 18;
```

Comment:
what would the ramifications be if a collateral currency was non-18 and did not implement this method? Your test with `BasicERC20` shows it correctly defaults to 18 but I feel that this will only let us use a subset of collateral types as some will not expose this method and NOT be 18 decimal delimited. I'm not sure if there are any tokens like this but I am pretty sure that it would make them incompatible with the perp as a collateral type.

Reply:
Yes it would make them incompatible unless we deployed a new factory that allowed us to hardcode the decimal precision.